src/QsCompiler/Transformations/TypeResolutionCombination.cs

```diff
@@ -0,0 +1,389 @@
+        /// IndependentResolutionDictionaries. Represents a combination of all the type parameters
+        /// found and their ultimate type resolutions.
+        /// </summary>
+        public TypeParameterResolutions CombinedResolutionDictionary { get; private set; }
+
+        /// <summary>
+        /// Flag for if there were any invalid scenarios encountered while creating the combination.
+        /// Invalid scenarios include a type parameter being assigned to multiple conflicting types
+        /// and a type parameter being assigned to a type referencing a different type parameter of
+        /// the same callable. Has value true if no invalid scenarios were encountered.
+        /// </summary>
+        public bool IsValid { get => !this.combinesOverConflictingResolution && !this.combinesOverParameterConstriction; }
+
+        /// <summary>
+        /// Flag for if, at any time in the creation of the combination, there was a type parameter that
+        /// was assigned conflicting type resolutions. Has value true if a conflict was encountered.
+        /// </summary>
+        private bool combinesOverConflictingResolution = false;
+
+        /// <summary>
+        /// Flag for if, at any time in the creation of the combination, there was a type parameter that
+        /// was assigned a type resolution referencing a different type parameter of the same callable.
+        /// </summary>
+        private bool combinesOverParameterConstriction = false;
+
+        // Constructors
+
+        /// <summary>
+        /// Creates a type parameter resolution combination from the independent type parameter resolutions
+        /// found in the given typed expression and its sub expressions. Only sub-expressions whose
+        /// type parameter resolutions are relevant to the given expression's type parameter resolutions
+        /// are considered.
+        /// </summary>
+        public TypeResolutionCombination(TypedExpression expression) : this(GetTypeParameterResolutions.Apply(expression))
+        {
+        }
+
+        /// <summary>
+        /// Creates a type parameter resolution combination from independent type parameter resolution dictionaries.
+        /// The given resolutions are expected to be ordered such that dictionaries containing type parameters resolutions that
+        /// reference type parameters in other dictionaries appear before those dictionaries containing the referenced type parameters.
+        /// I.e., dictionary A depends on dictionary B, so A should come before B. When using this method to resolve
+        /// the resolutions of a nested expression, this means that the innermost resolutions should come first, followed by
+        /// the next innermost, and so on until the outermost expression is given last.
+        /// </summary>
+        internal TypeResolutionCombination(params TypeParameterResolutions[] independentResolutionDictionaries)
```

Comment:
The doc comment should mention when something throws (it throws if any of the given dictionaries is null).

Reply:
I've added logic to filter out any null dictionaries instead of throwing on them and updated the doc comment to include that.